sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -285,6 +287,110 @@ case class ArrayTransform(
   override def prettyName: String = "transform"
 }
 
+/**
+ * Sorts elements in an array using a comparator function.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(expr, func) - Sorts and returns the array based on the given " +
+    "comparator function. The comparator will take two nullable arguments " +
+    "representing two nullable elements of the array." +
+    "It returns -1, 0, or 1 as the first nullable element is less than, equal to, or greater" +
+    "than the second nullable element. Null elements will be placed at the end of the returned" +
+    "array. If the comparator function returns other values (including NULL)," +
+    "the query will fail and raise an error. By the default it will sort the array in " +
+    "ascending mode",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(5, 6, 1), (x,y) -> f(x,y));
+       [1,5,6]
+      > SELECT _FUNC_(array('bc', 'ab', 'dc'), (x, y) -> f(x,y));
+       ['dc', 'bc', 'ab']
+  """,
+  since = "3.0.0")
+case class ArraySort(
+    argument: Expression,
+    function: Expression)
+  extends ArrayBasedSimpleHigherOrderFunction with ArraySortLike with CodegenFallback {
+
+  def this(argument: Expression) = this(argument, Literal(true))
+
+  @transient lazy val argumentsType: DataType =
+    argument.dataType.asInstanceOf[ArrayType].elementType
+
+  override protected def arrayExpression: Expression = argument
+
+  override protected def nullOrder: NullOrder = NullOrder.Greatest
+
+  override def dataType: ArrayType = ArrayType(argumentsType, argument.nullable)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+
+    if (function.dataType == BooleanType) {
```

Comment:
Does this mean you are trying to support `array_sort(column, order)` signature as well?

Reply:
I agree. `function` should not take `BooleanType`.